specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/cloudhsm.json

```diff
@@ -0,0 +1,1441 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Private endpoint connection successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnection"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "CloudHsmClusterPrivateEndpointConnection_Get_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmClusterPrivateEndpointConnection_Get_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/privateEndpointConnections": {
+      "get": {
+        "tags": [
+          "Cloud HSM Cluster Private Endpoint Connections - Get"
+        ],
+        "operationId": "PrivateEndpointConnections_ListByCloudHsmCluster",
+        "description": "The List operation gets information about the private endpoint connections associated with the Cloud HSM Cluster",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/cloudHsmClusterName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "List of all CloudHsmCluster private endpoint connections associated with the Cloud HSM Cluster.",
+            "schema": {
+              "$ref": "#/definitions/PrivateEndpointConnectionListResult"
+            }
+          },
+          "default": {
+            "description": "The error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v6/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "CloudHsmClusterPrivateEndpointConnection_ListByResource_MaximumSet_Gen": {
+            "$ref": "./examples/CloudHsmClusterPrivateEndpointConnection_ListByCloudHsmCluster_MaximumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.HardwareSecurityModules/cloudHsmClusters/{cloudHsmClusterName}/prebackup": {
```

Comment:
>prebackup [](http://example.com/codeflow?start=155&length=9)

Use a verb or verb phrase here, e.g. checkBackupAvailability, confirmBackupReadiness, validateBackupParameters, etc.

Reply:
We are following Azure Backup API convention, they are using PreBackup/PreRestore. https://msazure.visualstudio.com/One/_git/Mgmt-RecoverySvcs-WkloadCoord?path=/src/Documents/Swaggers/ArmBasedSwagger/ArmSwagger.json

Reply:
@mentat9 sound okay to you, or is that trumped by rbac guidelines?

Reply:
Thanks for the recommendation, after discussion with Azure Backup team, we can make a change to the API and used ValidateBackupPropeties/ValidateRestoreProperties